Interop/ModelCard/src/test/java/EnsembleModelsTest.java

```diff
@@ -0,0 +1,236 @@
+import org.tribuo.ensemble.WeightedEnsembleModel;
+import org.tribuo.math.optimisers.AdaGrad;
+import org.tribuo.multilabel.MultiLabel;
+import org.tribuo.multilabel.ensemble.MultiLabelVotingCombiner;
+import org.tribuo.multilabel.evaluation.MultiLabelEvaluator;
+import org.tribuo.multilabel.example.MultiLabelGaussianDataSource;
+import org.tribuo.multilabel.sgd.fm.FMMultiLabelTrainer;
+import org.tribuo.multilabel.sgd.objectives.BinaryCrossEntropy;
+import org.tribuo.regression.Regressor;
+import org.tribuo.regression.ensemble.AveragingCombiner;
+import org.tribuo.regression.evaluation.RegressionEvaluator;
+import org.tribuo.regression.example.NonlinearGaussianDataSource;
+import org.tribuo.regression.sgd.fm.FMRegressionTrainer;
+import org.tribuo.regression.sgd.objectives.SquaredLoss;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class EnsembleModelsTest {
+    // Classification combiners
+    private static final VotingCombiner VOTING = new VotingCombiner();
+    private static final FullyWeightedVotingCombiner FULL_VOTING = new FullyWeightedVotingCombiner();
+    // Regression combiners
+    private static final AveragingCombiner AVERAGING = new AveragingCombiner();
+    // Multi-label combiners
+    private static final MultiLabelVotingCombiner ML_VOTING = new MultiLabelVotingCombiner();
+
+    @BeforeAll
+    public static void setup() {
+        Class<?>[] classes = new Class<?>[]{
+                BaggingTrainer.class,
+                AbstractSGDTrainer.class,
+                org.tribuo.classification.sgd.linear.LinearSGDTrainer.class,
+                AbstractFMTrainer.class,
+                FMClassificationTrainer.class
+        };
+        for (Class<?> c : classes) {
+            Logger logger = Logger.getLogger(c.getName());
+            logger.setLevel(Level.WARNING);
+        }
+    }
+
+    @Test
+    public void testHomogenousClassificationModelCard() throws IOException {
+        DataSource<Label> trainSource = new NoisyInterlockingCrescentsDataSource(100,1,0.1);
+        MutableDataset<Label> train = new MutableDataset<>(trainSource);
+        DataSource<Label> testSource = new NoisyInterlockingCrescentsDataSource(100,2,0.1);
+        MutableDataset<Label> test = new MutableDataset<>(testSource);
+
+        LogisticRegressionTrainer lr = new LogisticRegressionTrainer();
+        BaggingTrainer<Label> t = new BaggingTrainer<>(lr, VOTING,5);
+        WeightedEnsembleModel<Label> ensemble = (WeightedEnsembleModel<Label>) t.train(train);
+
+        var evaluator = new LabelEvaluator();
+        var evaluation = evaluator.evaluate(ensemble, test);
+
+        // create Model Card object and write to file
+        ModelCard modelCard = new ModelCard(ensemble, evaluation);
+        modelCard.addMetric("overall-accuracy", evaluation.accuracy());
+        modelCard.saveToFile("src/test/output-json/homogeneousClassificationModelCard.json");
```

Comment:
This should use `File.createTempFile` and then set the file to `deleteOnExit`.

Reply:
Done as suggested.